core/trino-main/src/main/java/io/trino/sql/analyzer/RelationType.java

```diff
@@ -80,7 +79,6 @@ public int indexOf(Field field)
      */
     public Field getFieldByIndex(int fieldIndex)
     {
-        checkElementIndex(fieldIndex, allFields.size(), "fieldIndex");
         return allFields.get(fieldIndex);
```

Comment:
nit: I would keep it, for sake of nicer error message in case of failure.

Reply:
I am not sure if the error message is any better here.